### ISSUE TITLE
Feature Request: Ability to add Screen Lock to the dock

### ISSUE THREAD

**Issue by ibigpapa** [dangvd/ksmoothdock#68]
In Use an internal command or use a D-Bus command Screen Lock isn't available.  In the Application dock item it's available under session.  I would like to have as an icon on dock.

**Comment by dangvd** [comment]
You can just drag-and-drop the item in the Applications menu to the Edit Launchers dialog, or just edit the command manually with the Lock Screen command. The Use a D-Bus command drop-down is for convenience, any D-Bus command you type in would work.

Also, the reason I haven't added the Screen Lock to the drop down because of some minor look-and-feel issue.

**Comment by dangvd** [comment]
Ah sorry:
- You can't drag-and-drop the special items (Session/Power/Search) from the menu yet.
- Somehow using the Lock Screen D-Bus command from a Lock Screen dock item doesn't work. I manually created a dock item with the command "qdbus org.kde.screensaver /ScreenSaver Lock". I remember I tested it with KDE Plasma 4 and it worked (although with some minor look-and-feel issue) but with KDE Plasma 5 it doesn't. It does work from the Applications menu dock item though, so maybe the parabolic zooming effect or something interferes with it.

At the moment I'm not sure how to solve this.

**Comment by ibigpapa** [comment]
I did it a round about way.

created a script to lock the screen using loginctl

```python
#!/usr/bin/python3.6

import subprocess

info_task = subprocess.run(['loginctl', 'show-user', 'my_user'], stdout=subprocess.PIPE)

output = info_task.stdout.decode('utf-8')

sessions = [line for line in output.splitlines() if 'Sessions=' in line][0]

session_ids = sessions.replace('Sessions=','').split()

the_int = 1
for session_id in session_ids:
    try:
        the_int = int(session_id)
        continue
    except ValueError:
        pass

subprocess.run(['loginctl', 'lock-session', str(the_int)])
```

**Comment by dangvd** [comment]
Hi ibigpapa,
This has been fixed and released in v5.8
Hope you like it.
